shared/src/iosMain/kotlin/co/touchlab/kampstarter/models/Platform.kt

```diff
@@ -0,0 +1,31 @@
+package co.touchlab.kampstarter.models
+
+import co.touchlab.stately.concurrency.GuardedStableRef
+import kotlinx.coroutines.*
+import platform.Foundation.NSThread
+import kotlin.coroutines.CoroutineContext
+
+@OptIn(InternalCoroutinesApi::class)
+internal actual fun CoroutineScope.childContext(): CoroutineContext {
+    val ktorJob = Job()
+
+    val ref = GuardedStableRef(ktorJob)
+
+    val listenerDisposableHandle = coroutineContext[Job]!!.invokeOnCompletion(onCancelling = true) { cancelCause ->
+        val parentJob = ref.state
+
+        if (cancelCause is CancellationException)
+            parentJob.cancel(cause = cancelCause)
+        else
+            parentJob.cancel()
+    }
+
+    ktorJob.invokeOnCompletion {
+        listenerDisposableHandle.dispose()
+    }
+
+    return coroutineContext + ktorJob + coroutineContext[CoroutineExceptionHandler]!!
```

Comment:
Made some changes here. If you look back, `ktorJob` was actually not being used. That was my issue originally, not part of the refactor. Also, we weren't ever removing the `invokeOnCompletion` handler when the ktor stuff is done. I do think we need some kind of test, even if it's just run manually. To discuss.

Reply:
Works, though. At least happy path thest.